### ISSUE TITLE
Playing during navigation

### ISSUE THREAD

**Issue by gillesdutilh** [EmissarySocial/bandwagon#18]
I presume this is not trivial, but a functionality that would greatly improve the user experience would be a player that continues playing when you navigate across the website. 
Now, I realize that this is something that must become really challenging when you want to allow "federated navigation" while playing. I think that can only be done with an external player, right?

(by the way: Hi Ben, was great meeting you on the FediForum!)

**Comment by benpate** [comment]
Hey Giles!  Yeah, it was cool to meet at FediForum, glad we could connect :)

You're right, making a player that persists across pages is not trivial.  But fortunately, it's not impossible, either.  

Emissary uses the [htmx](https://htmx.org) library for all of its client-side, browser features.  The newest version (2.0) includes some features that should make this easier, when I can upgrade to it.  It'll also take rewriting some aspects of the player, but it needs a bunch of improvements as it is.  So right now, the list looks like:

- Persist player across page reloads
- Standardize, and clean up the way playlists are created (htmx + markup if possible)
- Allow player to be embedded in other pages
- Add "scrub bar" and volume controls
- Possibly support video content too?

I'm probably spending the next month or so on more logistical details, so I'm putting this further into the future for now.  But this is a great suggestion that I want to get to.  Thank you!

**Comment by gillesdutilh** [comment]
Sorry for the late reply, cool that you took this on your list. 
I think in the end, one would certainly want to have a player that works across servers, not only pages, right? In the ideal situation, one would have a bandwagon player (that may live in a browser window), that allows to browse any (other) bandwagon server, right?

**Comment by benpate** [comment]
Yes.  And this is an important part of the [discovery roadmap](https://bandwagon.fm/roadmap).

I'm not exactly sure of the architecture just yet.  I think it may be a separate page, or even a separate web service from a band's album pages.  So (like Apple Music) if you're on an album, you play only that album.  But if you go to some kind of library, radio station, or mix-tape, then you can play everything from that collection instead.  I think Funkwhale already does something like this.

I'd like to do it across servers, and even across platforms, bringing in (consenting) sites that use FairCamp, Funkwhale, Mirlo, or whoever else wants to join the party.  It's ambitious, so we'll have to see how much of this is possible.
